XamarinCommunityToolkit/Behaviors/EventToCommandBehavior.shared.cs

```diff
@@ -42,85 +44,56 @@ public object CommandParameter
-                throw new ArgumentException($"{nameof(EventToCommandBehavior)}: Can't register the {EventName} event.");
-
-            var methodInfo = typeof(EventToCommandBehavior).GetTypeInfo().GetDeclaredMethod(nameof(OnEvent));
-            eventHandler = methodInfo.CreateDelegate(eventInfo.EventHandlerType, this);
-            eventInfo.AddEventHandler(AssociatedObject, eventHandler);
-        }
-
-        void DeregisterEvent(string name)
+        void RegisterEvent()
         {
-            if (string.IsNullOrWhiteSpace(name) || eventHandler == null)
-                return;
+            UnregisterEvent();
 
-            var eventInfo = AssociatedObject.GetType().GetRuntimeEvent(name);
+            var eventName = EventName;
+            if (View == null || string.IsNullOrWhiteSpace(eventName))
+                return;
 
-            if (eventInfo == null)
-                throw new ArgumentException($"{nameof(EventToCommandBehavior)}: Can't de-register the {EventName} event.");
+            eventInfo = View.GetType().GetRuntimeEvent(eventName) ??
+                throw new ArgumentException($"{nameof(EventToCommandBehavior)}: Could't resolve the event.", nameof(EventName));
 
-            eventInfo.RemoveEventHandler(AssociatedObject, eventHandler);
-            eventHandler = null;
+            eventInfo.AddEventHandler(View, eventHandler);
         }
 
-        async void OnEvent(object sender, object eventArgs)
+        void UnregisterEvent()
         {
-            if (Command == null)
-                return;
-
-            var resolvedParameter = CommandParameter
-                ?? Converter.Convert(eventArgs, typeof(object), null, null)
-                ?? eventArgs;
-
-            if (Command.CanExecute(resolvedParameter))
-                Command.Execute(resolvedParameter);
-
-            foreach (var bindable in Actions)
-            {
-                bindable.BindingContext = BindingContext;
-                var action = (IAction)bindable;
-                await action.Execute(sender, eventArgs);
-            }
+            eventInfo?.RemoveEventHandler(View, eventHandler);
+            eventInfo = null;
         }
 
-        static void OnEventNameChanged(BindableObject bindable, object oldValue, object newValue)
+        void OnEventOccurred(object sender = null, object eventArgs = null)
         {
-            var behavior = (EventToCommandBehavior)bindable;
-
-            if (behavior.AssociatedObject == null)
-                return;
-
-            var oldEventName = (string)oldValue;
-            var newEventName = (string)newValue;
+            var parameter = CommandParameter
+                ?? EventArgsConverter?.Convert(eventArgs, typeof(object), null, null)
+                ?? eventArgs;
 
-            behavior.DeregisterEvent(oldEventName);
-            behavior.RegisterEvent(newEventName);
+            var command = Command;
+            if (command?.CanExecute(parameter) ?? false)
```

Comment:
What happens is `parameter == null` and the user did not specify any parameters on their command. Will this still work? Not sure how they solved that internally, maybe it will just ignore it because the value is null anyway. Is that something we can (unit) test?

Reply:
Yes, it will work fine. CanExecute should always take some parameter even though it's null

We can unit test it for sure.

Reply:
> Yes, it will work fine. CanExecute should always take some parameter even though it's null
> 
> We can unit test it for sure, but it won't fail (can execute management is users responsibility. We can't manage it)